core-navigation/src/main/kotlin/com/hedvig/android/navigation/Navigator.kt

```diff
@@ -0,0 +1,11 @@
+package com.hedvig.android.navigation
+
+import android.content.Context
+
+class Navigator(
```

Comment:
Would it make sense to add other navigation related code here? Like opening web urls, keyboard etc? Could keep that in mind :)

Reply:
Yeah in general there's definitely room to explore what we can do for navigation in our project.
In this case this was needed because I needed access to the `Context.startChat()` function but it lives inside the :app module, so :feature-odyssey couldn't see it.
Now, I could move that entire function to the newly created :core-navigation, but `startChat()` needs a reference to the ChatActivity because that's how it knows what to launch, and it doesn't make sense to extract that out of :app this way.
So this way, :app keeps the logic of how to navigate to the ChatActivity, by keeping the `startChat()` function in there, and sets up this `Navigator` by passing it to its lambda, while making this `Navigator` available to whichever other module needs to navigate to chat.
But yeah, since I am not 100% confident with what's the best solution to this, especially with Activity navigation as opposed to Compose-only navigation like we do inside of Odyssey, I think this is good enough for now.

Reply:
Makes total sense